### PR TITLE
this fixes the extraction of only the year requested #2187

### PR DESCRIPTION
**Pull request by robkooper**
The code will now only extract the year that the user requested and not all the years that are in the metfile.

## Review Time Estimate

- [ ] Immediately
- [ ] Within one week
- [X] When possible

## Types of changes

- [X] Bug fix (non-breaking change which fixes an issue) 
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My change requires a change to the documentation.
- [X] I have updated the CHANGELOG.md.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by robkooper** [comment]
@serbinsh please make sure I don't reintroduce the bug you fixed #2095

File: models/sipnet/R/model2netcdf.SIPNET.R [hunk 1]
```diff
@@ -74,15 +74,11 @@ model2netcdf.SIPNET <- function(outdir, sitelat, sitelon, start_date, end_date,
...
+  # get all years that we want data from
   year_seq <- seq(lubridate::year(start_date), lubridate::year(end_date))
-  if (length(year_seq) != num_years) {
-    PEcAn.logger::logger.severe("Date range specified in model2netcdf.SIPNET() function call and 
-                                total number of years in SIPNET output are not equal")
-  }
 
   # check that specified years and output years match
-  if (!all(simulation_years %in% year_seq)) {
+  if (!all(year_seq %in% simulation_years)) {
```
Comment (infotroph) [inline models/sipnet/R/model2netcdf.SIPNET.R]: :bug: :hammer: :100:

Comment (serbinsh) [reply models/sipnet/R/model2netcdf.SIPNET.R]: This wasnt a bug it was a check based on some older issues with met process resulting in incorrect SIPNET years being run.

Comment (serbinsh) [reply models/sipnet/R/model2netcdf.SIPNET.R]: before this check (which was done awhile ago) SIPENT runs could complete even if the wrong years or dates were simulated and this was meant to throw an consistency error.  But that was based on the older met process code where we had a lot of errors from incorrect BETY entries, etc.  But given we have been cleaning these up, I think this change should maintain a similar behavior.

**Comment by robkooper** [comment]
@serbinsh not sure if this will 100% work, but not sure the previous code was working in that case as well.

If I have met data from 2001-01-01 - 2015-07-31 and I request a run from 2015-04-01 - 2015-09-31 it would claim success, since the year 2015 is part of the actual run. It only checks for the years. I think if we want to do a full check it will need to check the output of the model generated.

I think a better place to catch this (not sure this is done) is to see if the met data covers the actual run data, this should be done I think in the met conversion (if it is not already done).

**Pull request merged**